enhancements/20240612-node-disk-manager-refactor.md

```diff
@@ -0,0 +1,67 @@
+# Node Disk Manager Refactor
+
+## Summary
+
+The Node Disk Manager is responsible for managing the raw device on the node. The current implementation is not flexible enough to support various storage backends. This enhancement is to refactor the Node Disk Manager to make it more flexible and extensible.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/6014
+
+## Motivation
+
+We need to refactor the current implementation to make the Node Disk Manager support various storage backends. This could help define the storage provisioner easily.
+
+### Goals
+
+Refactor the Node Disk Manager to make it more flexible and extensible to support various storage backends. The refactor will decompose the individual storage provisioner.
+
+### Non-goals [optional]
+
+None
+
```

Comment:
If there's no non-goals I think this section can be simply deleted.

Reply:
done!